libssh-rs/src/lib.rs

```diff
@@ -72,9 +72,12 @@ fn initialize() -> SshResult<()> {
 }
 
 pub(crate) struct SessionHolder {
+    outer: Weak<Mutex<SessionHolder>>,
```

Comment:
I'd lean towards naming this something like `self_ref`, and add a comment about how it is needed for callbacks that need to construct a Channel

Reply:
Done.